tcl/org.testeditor.tcl.dsl.ide.tests/src/org/testeditor/tcl/dsl/ide/TclContentProposalProviderTest.xtend

```diff
@@ -0,0 +1,370 @@
+			expect('confidential "param" into <element>').prefix.assertEquals('')
+			reject('Stop application') // proposal must NOT include templates with different prefix
+		]
+	}
+
+	@Test
+	def void testComponentIncompleteMultipartTemplateProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Start app|
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('application "path"').prefix.assertEquals('app')
+			reject('Stop application') // proposal must NOT include templates with different prefix
+		]
+	}
+
+	@Test
+	def void testComponentDenormalizedIncompleteMultipartTemplateProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			      - 	Start   	 app|
+		'''.withPackage
+	
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('application "path"').prefix.assertEquals('app')
+			reject('Stop application')
+		]
+	}
+
+	@Test
+	def void testElementProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <|>
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('Input').prefix.assertEquals('<')
+			expect('bar').prefix.assertEquals('<')
+			reject('Ok') // buttons have no 'Read value from' - fixture
+		]
+	}
+
+	@Test
+	def void testElementProposalNoClosingBracket() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <|
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			expect('Input>').prefix.assertEquals('<')
+			expect('bar>').prefix.assertEquals('<')
+			reject('Ok>') // buttons have no 'Read value from' - fixture
+		]
+	}
+
+	@Test
+	def void testIncompleteElementProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <b|>
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			reject('Input')
+			expect('bar').prefix.assertEquals('b')
+			reject('Ok') 
+		]
+	}
+
+	@Test
+	def void testTrailingElementProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Component: GreetingApplication
+			- Read value from <Input> |
+		'''.withPackage
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
+			reject('Input>')
+			reject('bar>')
+			reject('Ok>') 
+		]
+	}
+
+	@Test
+	def void testMacroTemplateProposal() {
+		// given
+		var tclSnippet = '''
+			# Some
+			
+			* some step
+			Macro: MacroLibrary
+			- |
+		'''.withPackage		
+
+		// when
+		val proposals = tclSnippet.proposals
+
+		// then
+		proposals => [
```

Comment:
Are there further proposals allowed here? If not, it might be more precise to assert that _exactly_ these proposals are present, and no additional ones.